src/node.rs

```diff
@@ -291,23 +425,43 @@ impl OscillatorNode {
         })
     }
 
+    /// Returns the oscillator frequency audio parameter
     pub fn frequency(&self) -> &AudioParam {
         &self.frequency
     }
 
+    /// Returns the oscillator type
     pub fn type_(&self) -> OscillatorType {
         self.type_.load(Ordering::SeqCst).into()
     }
 
+    /// set the oscillator type
     pub fn set_type(&self, type_: OscillatorType) {
         self.type_.store(type_ as u32, Ordering::SeqCst);
     }
+
+    /// set the oscillator type to custom. The oscillator will generate
+    /// a perdioc waveform following the PeriodicWave characteristics
+    //
+    //  TODO: The current implementation doesn't communicate its state
+    //  to the OscillatorRenderer, and so has no effect on the rendering.
+    //  This function should send the updated periodics waveform characteristics
+    //  to the OscillatorRenderer and more specifically to the CustomRenderer
+    pub fn set_periodic_wave(&mut self, _periodic_wave: PeriodicWave) {
+        self.set_type(OscillatorType::Custom);
+        todo!();
```

Comment:
You can use a rendez-vous channel. A bit more complicated setup is used in the AnalyserNode as that requires back and forth communication.
It sets up a `crossbeam_channel::bounded(0)` channel. The user-facing node can send the PeriodicWave (and will block), the render thread will come by and use `.try_recv()` (so not blocking) to check if any new wave needs to be picked up.